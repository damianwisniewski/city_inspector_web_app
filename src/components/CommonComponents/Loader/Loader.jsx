import React from 'react'
import PropTypes from 'prop-types'

import './Loader.scss'

const Loader = ({ id, size }) => {
	return <div id={id} className={`Loader Loader--${size}`} />
}

Loader.defaultProps = {
	size: 'medium',
}

Loader.propTypes = {
	size: PropTypes.oneOf(['small', 'medium', 'large']),
}

export default Loader
