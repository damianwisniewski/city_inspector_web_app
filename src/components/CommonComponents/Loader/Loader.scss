@import '../../../assets/styleHelpers/index';

.Loader {
	display: flex;
	justify-content: center;
	width: 100%;

	&::after {
		display: block;
		width: min-content;
		padding: 3px;

		color: $primary_light_blue;
		font-weight: 900;
		font-family: 'Font Awesome 5 Free';

		animation: rotate infinite 1.3s cubic-bezier(0.28, 0.81, 1, 1);

		content: '\f110';
	}

	&--small {
		font-size: 20px;
	}

	&--medium {
		font-size: 30px;
	}

	&--large {
		font-size: 45px;
	}
}

@keyframes rotate {
	0% {
		transform: rotate(0deg);
	}

	100% {
		transform: rotate(360deg);
	}
}
