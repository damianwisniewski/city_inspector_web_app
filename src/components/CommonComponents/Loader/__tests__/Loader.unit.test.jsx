import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

import Loader from '../Loader'

configure({ adapter: new Adapter() })

describe('<Loader />', () => {
	let LoaderElement
	const id = 'loaderTest'

	test('should render correctly Loader', () => {
		LoaderElement = shallow(<Loader id={id} />)
		expect(toJson(LoaderElement)).toMatchSnapshot()
	})

	test('should render correctly Loader width size "small"', () => {
		LoaderElement = shallow(<Loader id={id} size='small' />)
		expect(toJson(LoaderElement)).toMatchSnapshot()
	})

	test('should render correctly Loader width size "medium"', () => {
		LoaderElement = shallow(<Loader id={id} size='medium' />)
		expect(toJson(LoaderElement)).toMatchSnapshot()
	})

	test('should render correctly Loader width size "large"', () => {
		LoaderElement = shallow(<Loader id={id} size='large' />)
		expect(toJson(LoaderElement)).toMatchSnapshot()
	})
})
