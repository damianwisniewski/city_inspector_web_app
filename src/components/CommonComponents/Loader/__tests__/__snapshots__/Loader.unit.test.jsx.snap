// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should render correctly Loader 1`] = `
<div
  className="Loader Loader--medium"
  id="loaderTest"
/>
`;

exports[`<Loader /> should render correctly Loader width size "large" 1`] = `
<div
  className="Loader Loader--large"
  id="loaderTest"
/>
`;

exports[`<Loader /> should render correctly Loader width size "medium" 1`] = `
<div
  className="Loader Loader--medium"
  id="loaderTest"
/>
`;

exports[`<Loader /> should render correctly Loader width size "small" 1`] = `
<div
  className="Loader Loader--small"
  id="loaderTest"
/>
`;
